### PR TITLE
Test new eo staclike metadata

### PR DESCRIPTION
**Pull request by santoshamohan**
### Reason for this pull request
Index pilot test dataset for new metadata changes

### Proposed changes
- Add new function (`parse_dataset_field`) to extract dataset fields from the template
- Add `pytest` to use new `metadata` `eo` `staclike` template and index the new product

 - [x] Tests passed

### CODE REVIEW

File: integration_tests/conftest.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def ingest_configs(datacube_env_name):
     }
 
 
-@pytest.fixture(params=["US/Pacific", "UTC", ])
+@pytest.fixture(params=["UTC", ])
```
Comment (Kirill888) [inline integration_tests/conftest.py]: was this change intentional?

Comment (santoshamohan) [reply integration_tests/conftest.py]: Reverted the changes. Was pushed unintentionally.

File: integration_tests/test_new_eo_staclike_metadata_fields.py [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+
+EO_STACLIKE_METADATA_DOC = yaml.load('''---
+name: eo_stac_like_product
+description: Stack_datasets
+dataset:
+  id: [id]
+  creation_dt: ['properties', 'odc:processing_datetime']
+  label: [title]
+  measurements: [measurements]
+  grid_spatial: []
```
Comment (Kirill888) [inline integration_tests/test_new_eo_staclike_metadata_fields.py]: we need `grid_spatial` to point to some place real

Comment (santoshamohan) [reply integration_tests/test_new_eo_staclike_metadata_fields.py]: Updated tests to point `grid_spatial` to `geometric` reference points.

File: integration_tests/test_new_eo_staclike_metadata_fields.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    except AttributeError:
+        return str(val)
+
+
+def _get_boundingbox(metadata_fname):
+    with open(str(metadata_fname)) as fl:
+        metadata = yaml.load(fl)
+    crs = CRS(metadata['crs'])
+    geo = Geometry(metadata['geometry'], crs=crs)
+    spatial_reference = str(crs.wkt)
```
Comment (jeremyh) [inline integration_tests/test_new_eo_staclike_metadata_fields.py]: Use epsg code when available rather than wkt.

Eg.
```suggestion
    spatial_reference = 'epsg:{}'.format(crs.epsg) if crs.epsg else crs.wkt
```

File: integration_tests/test_new_eo_staclike_metadata_fields.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+    return xy_geo_ref_points, spatial_reference, lat_lon_coords
+
+
+def _create_new_dataset_def(input_file: Path):
+    """
+    Create new metadata dataset definition yaml files
+    """
+    now = datetime.utcnow()
+    creation_dt = now.strftime('%Y-%m-%dT%H:%M:%S.%f')
```
Comment (jeremyh) [inline integration_tests/test_new_eo_staclike_metadata_fields.py]: Creation time should refer to the "odc:processing_datetime" field. (it's when the data was created, not when the metadata was created)

File: integration_tests/test_new_eo_staclike_metadata_fields.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    xy_geo_ref_points, spatial_ref, lat_lon_coords = _get_boundingbox(input_file)
+
+    def _get_measurements(m):
+        m_dict = {}
+        for band, img_path in m.items():
+            m_dict[band] = {'path': str(img_path['path']), 'layer': 1}
+        return m_dict
+
+    dataset_def_template = {
+        'id': str(uuid.uuid4()),
```
Comment (jeremyh) [inline integration_tests/test_new_eo_staclike_metadata_fields.py]: Use the existing id from the dataset rather than creating a new one each time.

File: integration_tests/new-eo-staclike-metadata-type.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
+name: ga_ls5t_ard_3
+description: eo stack like
```
Comment (jeremyh) [inline integration_tests/new-eo-staclike-metadata-type.yaml:2]: The code isn't using this metadata type for indexing, only for extracting values from the input yaml. It may be simpler to just grab the fields directly, unless we have plans to use this elsewhere.

File: integration_tests/test_new_eo_staclike_metadata_fields.py [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+    clirunner(['-v', 'product', 'add', product_definition])
+
+    clirunner(['dataset', 'add', '--product', 'ga_ls5t_ard_3', str(dataset_definition)])
+
+    datasets = index.datasets.search_eager(product='ga_ls5t_ard_3')
+    assert len(datasets) > 0
+    assert not datasets[0].managed
+
+    ds_ = index.datasets.get(_ds['id'], include_sources=True)
+    assert ds_ is not None
```
Comment (jeremyh) [inline integration_tests/test_new_eo_staclike_metadata_fields.py:241]: Add some asserts to check that the `EXPECTED_VALUE` fields are still the same after copying.

```
>>> ds.metadata.time
None
```

File: integration_tests/test_new_eo_staclike_metadata_fields.py [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+    def _get_measurements(m):
+        m_dict = {}
+        for band, img_path in m.items():
+            m_dict[band] = {'path': str(img_path['path']), 'layer': 1}
+        return m_dict
+
+    dataset_def_template = {
+        'id': str(uuid.uuid4()),
+        'creation_dt': creation_dt,
+        'label': '',
```
Comment (jeremyh) [inline integration_tests/test_new_eo_staclike_metadata_fields.py]: The new-style metadata has a label in the naming conventions, so we could:
```suggestion
        'label': input_file.name.split('.')[0],
```
(or otherwise: it's better to remove it than use the blank string)

File: integration_tests/test_new_eo_staclike_metadata_fields.py [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+    lat=None,
+    lon=None,
+    time=Range(datetime.strptime('2009-12-17 23:53:03.946464', "%Y-%m-%d %H:%M:%S.%f"),
+               datetime.strptime('2009-12-17 23:53:31.463325', "%Y-%m-%d %H:%M:%S.%f")),
+)
+
+
+def mk_measurement(m):
+    common = dict(dtype='int16',
+                  nodata=-999,
```
Comment (jeremyh) [inline integration_tests/test_new_eo_staclike_metadata_fields.py:34]: Some bands aren't `-999`/`int16`. We'll probably need a map.

File: integration_tests/test_new_eo_staclike_metadata_fields.py [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+                                             'oa_azimuthal_incident', 'oa_combined_terrain_shadow', 'oa_exiting_angle',
+                                             'oa_fmask', 'oa_incident_angle', 'oa_nbar_contiguity',
+                                             'oa_nbart_contiguity', 'oa_relative_azimuth', 'oa_relative_slope',
+                                             'oa_satellite_azimuth', 'oa_satellite_view', 'oa_solar_azimuth',
+                                             'oa_solar_zenith', 'oa_time_delta'),
+                               with_grid_spec=False,
+                               storage=None):
+
+    if storage is None and with_grid_spec is True:
+        storage = {'crs': 'epsg:4326'}
```
Comment (jeremyh) [inline integration_tests/test_new_eo_staclike_metadata_fields.py]: This doesn't seem right -- we'll need to discuss/decide

**Review by jeremyh — COMMENTED** [review]
Nice work Santosh.

Users want many of the `cloud_cover`/`gqa`/`fmask` fields indexed too, so we might want to add them to the output. Perhaps the simplest option is to copy the whole `properties` section into the converted file? Or else we pick and choose them individually.

**Comment by Kirill888** [comment]
@santoshamohan adn @jeremyh I think we should just close this without merging. All the metadata related work is in other repos.

**Comment by santoshamohan** [comment]
Agree with @Kirill888
